cql3/statements/select_statement.cc

```diff
@@ -559,7 +560,9 @@ indexed_table_select_statement::do_execute_base_query(
     auto cmd = prepare_command_for_base_query(qp, options, state, now, bool(paging_state));
     auto timeout = db::timeout_clock::now() + get_timeout(state.get_client_state(), options);
     uint32_t queried_ranges_count = partition_ranges.size();
-    query_ranges_to_vnodes_generator ranges_to_vnodes(qp.proxy().get_token_metadata_ptr(), _schema, std::move(partition_ranges));
+    auto&& table = qp.proxy().local_db().find_column_family(_schema);
+    auto erm = table.get_effective_replication_map();
+    query_ranges_to_vnodes_generator ranges_to_vnodes(erm->make_splitter(), _schema, std::move(partition_ranges));
```

Comment:
There's an existing bug where erm is not captured for the duration of the page. So if a query is slower than ring_delay, and streaming is too fast (both very unlikely) then we can query bad data.

Reply:
I anticipated that and token_range_splitter survives erm by keeping token_metadata_ptr alive.

Reply:
yes it will keep erm, but holding erm right now doesn't prevent the real topology from moving. So we have to trust ring_delay for now.

Reply:
Ah, I said "erm is not captured". Well capturing it prepares the way for the full solution.